py27googledriveconnect/googledriveutils.py

```diff
@@ -0,0 +1,164 @@
+"""
+Written By: Kathryn Cogert
+For: Winkler Lab/CSE599 Winter Quarter 2016
+Purpose: Write and read from a google doc
+
+To Do:
++Add appending data to existing files functions (write now everything just overwrites)
++Add exporting from cRIO portion
+
+
+Note: Had to copy of local copy of PyDrive because it has some issues that make it incompatible with the latest
+version of the googleAPI.  This copy is in it's own directory and
+Specifically, it forces a / at the end of the redirect URI.  Google does not allow a slash at the end of those URLS
```

Comment:
Google calls it a "Redirect URI" (that's right, I not L). I had no idea what that was and didn't really bother to think about it until this comment.  But just looked it up:

URI = Unique Resource Identifier
-refers to any string of characters that can identify a resource
URL = Unique Resource Locator
-refers to a string of characters that **tell you where to find that resource**
URN = " " Name
-refers to a string that identifies a name in a specific name space.

So for example, you have several ways to identify yourself.  Your URN (your name Michael Newton, given in FirstName LastName format and listed in our class list or whatever) or your URL (1122 Boogie Woogie Avenue)

So what does that have to do with google drive APIs?  Well the redirect URI is where the client (read: the thing that is using the program) will get sent to after the account authorization is successful.  And we're sending them to a local host, which exists only locally to the client (http://localhost:8080).  That local host is not a URL (and this is where I get a little iffy on all this) because I don't believe it's describing a specific location.  It's just the identity of a resource with not real location attached to it.  

So long story as to why I didn't address this one.